tracks/python/exercises/grains/mentoring.md

```diff
@@ -0,0 +1,65 @@
+# Reasonable solution by calculating power.
+
+```python
+def square(number):
+    """
+    This function must return the number of grains on a given square number.
+    :param: int square number on which we want to calculate the amount of grains.
+    :return int amount of grains present on the give square number(number).
+    """
+    if number < 1 or number > 64:
+        raise ValueError("square must be between 1 and 64")
+    else:
+        return 2**(number - 1)
+
+def total():
+    """
+    This function must return the total number of grains on the chess board.
+    :return int total number of grains a chess board can contain.
+    """
+    # As 2^0 + 2^1 + 2^2 + 2^3 +....+ 2^n = (2^(n + 1)) - 1.
+    return 2**64 - 1
+```
+
+## For function `square(number)` :
+
+As we did `2**(number - 1)` on block one there will be only 1 grain, 2 to the power of 0 equals 1.
+On block 2 there will 2 and on block 3 there will be 4 which is equal to 2 to the power of 1, and 2 to the power of 2, respectively.
+Hence generalized, **2 to the power of (number-1)**.
+
+## For function `total()` :
+
+There is a general formula : 
```

Comment:
This is the first place that there is trailing white space.  Here is a diff that can be applied to address the rest of them as well:

```diff
diff --git i/tracks/python/exercises/grains/mentoring.md w/tracks/python/exercises/grains/mentoring.md
index 3e7e7ab2..5696c77a 100644
--- i/tracks/python/exercises/grains/mentoring.md
+++ w/tracks/python/exercises/grains/mentoring.md
@@ -29,7 +29,7 @@ Hence generalized, **2 to the power of (number-1)**.
 
 ## For function `total()` :
 
-There is a general formula : 
+There is a general formula :
 **2^0 + 2^1 + 2^2 + 2^3 +....+ 2^n = (2^(n + 1)) - 1**.
 
 # Solution using bitwise left operator.
@@ -39,8 +39,8 @@ def square(number):
     if number < 1 or number > 64:
         raise ValueError("square must be between 1 and 64")
     else:
-        return 1 << number - 1 
-        
+        return 1 << number - 1
+
 def total():
     # As 2^0 + 2^1 + 2^2 + 2^3 +....+ 2^n = (2^(n + 1)) - 1
     return (1 << 64) - 1
@@ -48,7 +48,7 @@ def total():
 
 # Left shift bitwise operator.
 
-**2 to the power of n = 1 << n** 
+**2 to the power of n = 1 << n**
 ```python
 >>> 2^0
 1
```

Reply:
done